client/components/users/UserQuizHistoryTaken.jsx

```diff
@@ -65,88 +98,56 @@ function UserQuizHistoryTaken(userId) {
 
   useEffect(() => {
     retrieveQuizHistory();
+    refreshFriends();
+    return () => {
+      setFriends([])
+    }
   }, [])
```

Comment:
Wouldn't you want this useEffect to fire whenever a new user with their own list of friends login? I think this only fires in the initial rendering, so that the friend list will always be the same no matter what user is logged in.

Reply:
I see what you're saying, and I added a console.log inside this useEffect to see when it fires to test it out. This useEffect only fires after you click the "Quizzes Taken" tab inside the quiz history modal, and not when the User Profile or Home page renders. I also tried switching to a different profile and checking out the list of friends, and they were different so it is confirmed that this useEffect fires for every new user. Hope that clears things up. 

Additionally, this same useEffect exists on lines 28-33 of QuizHistoryDisplay, that fires on initial rendering as well!

Reply:
Ahhh gotcha